satellite/payments/coinpayments/transactions.go

```diff
@@ -0,0 +1,320 @@
+func (tx *Transaction) UnmarshalJSON(b []byte) error {
+	var txRaw struct {
+		Amount         string `json:"amount"`
+		Address        string `json:"address"`
+		DestTag        string `json:"dest_tag"`
+		TxID           string `json:"txn_id"`
+		ConfirmsNeeded string `json:"confirms_needed"`
+		Timeout        int    `json:"timeout"`
+		CheckoutURL    string `json:"checkout_url"`
+		StatusURL      string `json:"status_url"`
+		QRCodeURL      string `json:"qrcode_url"`
+	}
+
+	if err := json.Unmarshal(b, &txRaw); err != nil {
+		return err
+	}
+
+	amount, err := strconv.ParseFloat(txRaw.Amount, 64)
+	if err != nil {
+		return err
+	}
+	confirms, err := strconv.ParseInt(txRaw.ConfirmsNeeded, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	*tx = Transaction{
+		ID:             TransactionID(txRaw.TxID),
+		Address:        txRaw.Address,
+		Amount:         amount,
+		DestTag:        txRaw.DestTag,
+		ConfirmsNeeded: int(confirms),
+		Timeout:        time.Second * time.Duration(txRaw.Timeout),
+		CheckoutURL:    txRaw.CheckoutURL,
+		StatusURL:      txRaw.StatusURL,
+		QRCodeURL:      txRaw.QRCodeURL,
+	}
+
+	return nil
+}
+
+// TransactionInfo holds transaction information.
+type TransactionInfo struct {
+	Address          string
+	Coin             Currency
+	Amount           float64
+	Received         float64
+	ConfirmsReceived int
+	Status           Status
+	ExpiresAt        time.Time
+	CreatedAt        time.Time
+}
+
+// UnmarshalJSON handles json unmarshaling for transaction info.
+func (info *TransactionInfo) UnmarshalJSON(b []byte) error {
+	var txInfoRaw struct {
+		Address      string `json:"payment_address"`
+		Coin         string `json:"coin"`
+		Status       int    `json:"status"`
+		AmountF      string `json:"amountf"`
+		ReceivedF    string `json:"receivedf"`
+		ConfirmsRecv int    `json:"recv_confirms"`
+		ExpiresAt    int64  `json:"time_expires"`
+		CreatedAt    int64  `json:"time_created"`
+	}
+
+	if err := json.Unmarshal(b, &txInfoRaw); err != nil {
+		return err
+	}
+
+	amount, err := strconv.ParseFloat(txInfoRaw.AmountF, 64)
+	if err != nil {
+		return err
+	}
+	received, err := strconv.ParseFloat(txInfoRaw.ReceivedF, 64)
+	if err != nil {
+		return err
+	}
+
+	*info = TransactionInfo{
+		Address:          txInfoRaw.Address,
+		Coin:             Currency(txInfoRaw.Coin),
+		Amount:           amount,
+		Received:         received,
+		ConfirmsReceived: txInfoRaw.ConfirmsRecv,
+		Status:           Status(txInfoRaw.Status),
+		ExpiresAt:        time.Unix(txInfoRaw.ExpiresAt, 0),
+		CreatedAt:        time.Unix(txInfoRaw.CreatedAt, 0),
+	}
+
+	return nil
+}
+
+// TransactionInfos is map of transaction infos by transaction id.
+type TransactionInfos map[TransactionID]TransactionInfo
+
+// UnmarshalJSON handles json unmarshaling for TransactionInfos.
+func (infos *TransactionInfos) UnmarshalJSON(b []byte) error {
+	var _infos map[TransactionID]TransactionInfo
+
+	var errors map[TransactionID]struct {
+		Error string `json:"error"`
+	}
```

Comment:
You should be able to do a single parse when using:

```
var errors map[TransactionID]struct {
    TransactionInfo
    Error string `json:"error"`
}
```

Reply:
Unfortunately it doesn’t work like this, as we have custom `unmarshal` for this type